indra/llcorehttp/httprequest.h

```diff
@@ -217,6 +217,10 @@ class HttpRequest
         /// Per-class only
         PO_PIPELINING_DEPTH,
 
+        /// If it is 1, enable HTTP Multiplexing.
```

Comment:
More explanation about this API.  The 'public' API presented here is *not* meant to be a facade for libcurl.  At the time this was written and also possibly for future purposes, libcurl might be replaced by another library.  (For Sansar, we had our own low-level libcurl-like library which could have slotted in.)  The 'public' API avoided using libcurl definitions.

Instead, this API is meant to describe how the viewer wants to use HTTP requests and how HTTP traffic should be managed and shaped.  The intent is to have this description be as minimal as possible.  If a concept isn't used by the viewer, it won't appear here.

Internal APIs, such as _httplibcurl.*, are the translation layer between the public API and whatever is going on below.  An adapter layer.  There's only one concrete adapter at this point but that is okay.

And so we have the comment above:
> When PIPELINING_DEPTH is 2 or more, libcurl performs

That is still correct.  The public API describes an abstract usage and a pipelining depth of 2 or more enables pipelining.  And so the code above in _httplibcurl is no longer correct.

What *should* this API be in the http/2 world?  Well, we don't know yet.  We haven't stood up the infrastructure to support it.  But we would probably want to modify llcorehttp differently.  PO_PIPELINING_DEPTH might instead be changed to PO_REQUEST_CONCURRENCY covering both the http/1 (pipelining) and http/2 (multiplexing) cases with added explicit enables/disables for pipelining and http/2.  However, we may find that we need independent http/2 parameters for the policy classes.

(There's another weak reason to keep existing API definitions as-is:  we use a copy of this code in server-side binaries and subtle changes may not be caught.  This isn't meant to stop changes and progress in the viewer.  But when all other considerations are equal, keeping an API spec has a benefit to us.)